pyproject.toml

```diff
@@ -27,6 +27,7 @@ scikit-learn = "^1.5.1"
 ipykernel = "^6.29.5"
 typing-extensions = "^4.12.2"
 ipython = "^8.27.0"
+pytest = "^8.3.5"
```

Comment:
Since pytest is used with testing and not in the module code itself, let's prefer to add it to the development dependencies group.

Let's update the toml file using these commands:

```sh
poetry remove pytest 
poetry add --dev pytest
```

Then in the README let's update the installation command number 4 to be `poetry install --with dev`, which will install dependencies from the dev group as well.